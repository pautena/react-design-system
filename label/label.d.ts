import { SxProps, Theme } from '@mui/material/styles';
export type LabelVariant = "primary" | "secondary" | "default" | "info" | "warning" | "error" | "success";
export declare const labelClasses: {
    root: string;
};
export interface LabelProps {
    /**
     * Content of the component
     */
    text: string;
    /**
     * Color palette used to draw the component
     */
    variant?: LabelVariant;
    /**
     * Background color. Overrides variant
     */
    color?: string;
    /**
     * Custom styles
     */
    sx?: SxProps<Theme>;
    /**
     * Show the text as uppercase
     */
    textTransform?: "none" | "capitalize" | "uppercase";
}
/**
 * Compact element to represent a text
 */
export declare const Label: ({ text, color: colorProp, variant, textTransform, sx, }: LabelProps) => import("react/jsx-runtime").JSX.Element;
