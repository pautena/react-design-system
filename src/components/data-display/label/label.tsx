import { Box, SxProps, Theme, Typography, useTheme } from "@mui/material";
import React from "react";

export type LabelVariant = "primary" | "secondary" | "default" | "info" | "warning" | "error";

export const labelClasses = {
  root: "RdsLabel-root",
};

export interface LabelProps {
  /**
   * Content of the component
   */
  text: string;
  /**
   * Color palette used to draw the component
   */
  variant?: LabelVariant;

  /**
   * Custom styles
   */
  sx?: SxProps<Theme>;
}

/**
 * Compact element to represent a text
 */
export const Label = ({ text, variant = "default", sx }: LabelProps) => {
  const { palette } = useTheme();

  const backgroundColor: Record<LabelVariant, string> = {
    default: palette.mode === "light" ? palette.grey[100] : palette.grey[900],
    primary: palette.primary.main,
    secondary: palette.secondary.main,
    info: palette.info.main,
    warning: palette.warning.main,
    error: palette.error.main,
  };

  const textColor: Record<LabelVariant, string> = {
    default: palette.getContrastText(backgroundColor.default),
    primary: palette.primary.contrastText,
    secondary: palette.secondary.contrastText,
    info: palette.info.contrastText,
    warning: palette.warning.contrastText,
    error: palette.error.contrastText,
  };

  return (
    <Box
      px={1}
      sx={{ backgroundColor: backgroundColor[variant], ...sx }}
      borderRadius={1}
      color={textColor[variant]}
      className={labelClasses.root}
      role="label"
      aria-describedby={variant}
    >
      <Typography variant="caption" fontWeight={700}>
        {text.toUpperCase()}
      </Typography>
    </Box>
  );
};
