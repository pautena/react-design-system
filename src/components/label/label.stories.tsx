import { ComponentMeta } from "@storybook/react";
import { createTemplate } from "../../storybook";
import { Label } from "./label";

export default {
  title: "Data Display/Label",
  component: Label,
  parameters: {
    layout: "centered",
  },
} as ComponentMeta<typeof Label>;

const Template = createTemplate(Label);

export const Default = Template.bind({});
Default.args = {
  text: "lorem",
  variant: "default",
};

export const Primary = Template.bind({});
Primary.args = {
  text: "lorem",
  variant: "primary",
};

export const Secondary = Template.bind({});
Secondary.args = {
  text: "lorem",
  variant: "secondary",
};

export const Info = Template.bind({});
Info.args = {
  text: "lorem",
  variant: "info",
};

export const Warning = Template.bind({});
Warning.args = {
  text: "lorem",
  variant: "warning",
};

export const Error = Template.bind({});
Error.args = {
  text: "lorem",
  variant: "error",
};
