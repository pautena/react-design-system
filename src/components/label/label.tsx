import { Box, Typography, useTheme } from "@mui/material";
import React from "react";

export type LabelVariant = "primary" | "secondary" | "default" | "info" | "warning" | "error";

export const labelClasses = {
  root: "RdsLabel-root",
};

export interface LabelProps {
  text: string;
  variant?: LabelVariant;
}

export const Label = ({ text, variant = "default" }: LabelProps) => {
  const { palette } = useTheme();

  const backgroundColor: Record<LabelVariant, string> = {
    default: palette.mode === "light" ? palette.grey[100] : palette.grey[900],
    primary: palette.primary.main,
    secondary: palette.secondary.main,
    info: palette.info.main,
    warning: palette.warning.main,
    error: palette.error.main,
  };

  const textColor: Record<LabelVariant, string> = {
    default: palette.getContrastText(backgroundColor.default),
    primary: palette.primary.contrastText,
    secondary: palette.secondary.contrastText,
    info: palette.info.contrastText,
    warning: palette.warning.contrastText,
    error: palette.error.contrastText,
  };

  return (
    <Box
      px={1}
      sx={{ backgroundColor: backgroundColor[variant] }}
      borderRadius={1}
      color={textColor[variant]}
      className={labelClasses.root}
    >
      <Typography variant="caption" fontWeight={700}>
        {text.toUpperCase()}
      </Typography>
    </Box>
  );
};
