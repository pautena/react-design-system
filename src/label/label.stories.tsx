import { Meta, StoryObj } from "@storybook/react";
import { Label } from "./label";

export default {
  title: "Components/Data Display/Label",
  component: Label,
  parameters: {
    layout: "centered",
  },
} satisfies Meta<typeof Label>;
type Story = StoryObj<typeof Label>;

export const Default: Story = {
  args: {
    text: "lorem",
    variant: "default",
  },
};

export const Primary: Story = {
  args: {
    text: "lorem",
    variant: "primary",
  },
};

export const Secondary: Story = {
  args: {
    text: "lorem",
    variant: "secondary",
  },
};

export const Info: Story = {
  args: {
    text: "lorem",
    variant: "info",
  },
};

export const Warning: Story = {
  args: {
    text: "lorem",
    variant: "warning",
  },
};

export const Error: Story = {
  args: {
    text: "lorem",
    variant: "error",
  },
};
