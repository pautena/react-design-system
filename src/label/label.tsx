import Box from "@mui/material/Box";
import { SxProps, Theme, useTheme } from "@mui/material/styles";

export type LabelVariant =
  | "primary"
  | "secondary"
  | "default"
  | "info"
  | "warning"
  | "error"
  | "success";

export const labelClasses = {
  root: "RdsLabel-root",
};

export interface LabelProps {
  /**
   * Content of the component
   */
  text: string;
  /**
   * Color palette used to draw the component
   */
  variant?: LabelVariant;

  /**
   * Custom styles
   */
  sx?: SxProps<Theme>;

  /**
   * Show the text as uppercase
   */
  textTransform?: "none" | "capitalize" | "uppercase";
}

/**
 * Compact element to represent a text
 */
export const Label = ({
  text,
  variant = "default",
  textTransform = "capitalize",
  sx,
}: LabelProps) => {
  const { palette, typography } = useTheme();

  const backgroundColor: Record<LabelVariant, string> = {
    default: palette.mode === "light" ? palette.grey[100] : palette.grey[900],
    primary: palette.primary.light,
    secondary: palette.secondary.light,
    info: palette.info.light,
    warning: palette.warning.light,
    error: palette.error.light,
    success: palette.success.light,
  };

  const textColor: Record<LabelVariant, string> = {
    default: palette.getContrastText(backgroundColor.default),
    primary: palette.primary.contrastText,
    secondary: palette.secondary.contrastText,
    info: palette.info.contrastText,
    warning: palette.warning.contrastText,
    error: palette.error.contrastText,
    success: palette.success.contrastText,
  };

  return (
    <Box
      height={24}
      minWidth={22}
      display="inline-flex"
      justifyContent="center"
      alignItems="center"
      bgcolor={backgroundColor[variant]}
      color={textColor[variant]}
      fontSize={typography.caption.fontSize}
      fontWeight={typography.fontWeightBold}
      lineHeight={0}
      textTransform={textTransform}
      whiteSpace="nowrap"
      borderRadius={2}
      role="label"
      aria-label={`${text} ${variant} label`}
      py={0}
      px={1}
      sx={{ cursor: "default", ...sx }}
    >
      {text}
    </Box>
  );
};
