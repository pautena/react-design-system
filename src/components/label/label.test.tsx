import React from "react";
import { Label, LabelVariant } from "./label";
import { render, screen } from "../../tests";
import { PaletteMode } from "@mui/material";

describe("Label", () => {
  const renderComponent = (variant: LabelVariant | undefined = undefined) => {
    return render(<Label variant={variant} text="lorem ipsum" />);
  };

  it("renders the label text", () => {
    renderComponent();

    expect(screen.getByText("LOREM IPSUM")).toBeInTheDocument();
  });

  it.each([["primary"], ["secondary"], ["default"], ["info"], ["warning"], ["error"]])(
    "renders correctly with variant %s",
    (variant: string) => {
      const { container } = renderComponent(variant as LabelVariant);

      expect(screen.getByLabelText(variant)).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    },
  );
});
