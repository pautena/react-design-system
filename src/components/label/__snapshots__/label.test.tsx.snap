// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly with variant default 1`] = `
<div>
  <div
    aria-label="default"
    class="RdsLabel-root MuiBox-root css-125klzf"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-sfqsx-MuiTypography-root"
    >
      LOREM IPSUM
    </span>
  </div>
</div>
`;

exports[`Label renders correctly with variant error 1`] = `
<div>
  <div
    aria-label="error"
    class="RdsLabel-root MuiBox-root css-hr30l5"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-sfqsx-MuiTypography-root"
    >
      LOREM IPSUM
    </span>
  </div>
</div>
`;

exports[`Label renders correctly with variant info 1`] = `
<div>
  <div
    aria-label="info"
    class="RdsLabel-root MuiBox-root css-xsokld"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-sfqsx-MuiTypography-root"
    >
      LOREM IPSUM
    </span>
  </div>
</div>
`;

exports[`Label renders correctly with variant primary 1`] = `
<div>
  <div
    aria-label="primary"
    class="RdsLabel-root MuiBox-root css-1kqxb18"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-sfqsx-MuiTypography-root"
    >
      LOREM IPSUM
    </span>
  </div>
</div>
`;

exports[`Label renders correctly with variant secondary 1`] = `
<div>
  <div
    aria-label="secondary"
    class="RdsLabel-root MuiBox-root css-akwe60"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-sfqsx-MuiTypography-root"
    >
      LOREM IPSUM
    </span>
  </div>
</div>
`;

exports[`Label renders correctly with variant warning 1`] = `
<div>
  <div
    aria-label="warning"
    class="RdsLabel-root MuiBox-root css-pe0lg2"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-sfqsx-MuiTypography-root"
    >
      LOREM IPSUM
    </span>
  </div>
</div>
`;
